### PR TITLE
Reorganizing training logic to clear backend

### PR DESCRIPTION
**Pull request by Meadosc**
Reorganize training logic to clear backend when checkpoint is reached

### CODE REVIEW

File: pix2pix/main.py [hunk 1]
```diff
@@ -254,10 +248,50 @@ def discriminator(summary=False):
...
+                inter_op_parallelism_threads=num_sockets,
+            )
+        )
+    )
 ###########################################################
 # Training
 ###########################################################
-def train(model="checkpoint", cycles=1000, size_batch=50):
+def train(model="checkpoint", cycles=1000, size_batch=50, checkpoint=20):
+    intel_optimizations(first=True)
```
Comment (unrahul) [inline pix2pix/main.py]: change to `cpu_config"

File: pix2pix/main.py [hunk 2]
```diff
@@ -254,10 +248,50 @@ def discriminator(summary=False):
...
+        gen_model = generator()
+    try:
+        disc_model_file = max(discriminators, key=os.path.getctime)
+        disc_model = load_model(disc_model_file)
+    except ValueError:
+        disc_model = discriminator()
+
+    return disc_model, gen_model
+
+def intel_optimizations(first=False):
```
Comment (unrahul) [inline pix2pix/main.py]: `cpu_config`

File: pix2pix/scripts/helper.py [hunk 3]
```diff
@@ -4,14 +4,14 @@
...
-from glob import glob
+import glob
 from os.path import abspath, dirname
 import psutil
 import subprocess
 
 from PIL import Image
 from sklearn.preprocessing import normalize
-from tensorflow.keras.models import Model
+from tensorflow.keras.models import Model, load_model
```
Comment (unrahul) [inline pix2pix/scripts/helper.py]: load_model does not seem to be used.

File: pix2pix/main.py [hunk 4]
```diff
@@ -273,113 +307,111 @@ def train(model="checkpoint", cycles=1000, size_batch=50):
...
+        opt_gen = Adam(lr=LEARNING_RATE_GEN, beta_1=0.5, name="opt_adv_adam")
+        g_train.compile(
+            optimizer=opt_gen,
+            loss=["binary_crossentropy", "mae"],
+            metrics=["accuracy"],
+            loss_weights=[1, 100],
         )
-        try:
-            gen_model_file = max(generators, key=os.path.getctime)
-            gen_model = load_model(gen_model_file)
```
Comment (unrahul) [inline pix2pix/main.py]: nice, things should go a bit faster..

**Comment by Meadosc** [comment]
@unrahul Updates made per request

**Pull request merged**